src/Notes/WC_Admin_Notes_First_Order.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * WooCommerce Admin First Order Note.
+ *
+ * Adds a note on first order creation.
+ *
+ * @package WooCommerce Admin
+ */
+
+namespace Automattic\WooCommerce\Admin\Notes;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Notes_First_Order
+ */
+class WC_Admin_Notes_First_Order {
+	/**
+	 * Note traits.
+	 */
+	use NoteTraits;
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-admin-first-order';
+
+	/**
+	 * Attach hooks.
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_checkout_create_order', array( __CLASS__, 'possibly_add_note' ) );
+	}
+
+	/**
+	 * Get the note.
+	 */
+	public static function get_note() {
+		// Only add this note if no previous orders exist.
+		$query  = new \WC_Order_Query(
+			array(
+				'limit' => 1,
+			)
+		);
+		$orders = $query->get_orders();
+
+		if ( count( $orders ) > 0 ) {
```

Comment:
Doesn't the query include the newly created order in the `woocommerce_checkout_create_order` action? I would actually be surprised if that was the case as I saw similar behaviour when querying for products on add.

Reply:
I thought the same thing at first, but checking the code shows this fires right before save.  Confirmed on a new site that it works as expected, but this will change anyway as of https://github.com/woocommerce/woocommerce-admin/pull/4531/commits/327f693da60ce0555ad322d56b7f8057f7e77ae5 to check for non checkout orders as well.